### PR TITLE
G8driver refreshtoken

### PR DESCRIPTION
**Pull request by grimpy**
### Description

Update g8 driver fixed some small issues when creating portforwards
And adds functionality to update the apitoken

### Status
- done, ready for review

### Checklist (tick everything that applies)

- [ ] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)

### CODE REVIEW

File: libcloud/common/gig_g8.py [hunk 1]
```diff
@@ -26,6 +35,74 @@ def add_default_headers(self, headers):
...
+    :rtype: str
+    """
+    rem = len(input) % 4
+
+    if rem > 0:
+        input += b'=' * (4 - rem)
+    return base64.urlsafe_b64decode(input)
+
+
+def is_jwt_expired(jwt):
```
Comment (Kami) [inline libcloud/common/gig_g8.py:60]: Can you please add some unit tests for this function (covering both expired and non expired scenario, various forms of input string, etc.).

Comment (grimpy) [reply libcloud/common/gig_g8.py:60]: I added a test for expire and none expired scenarios, not sure what to do about 'forms of input string' the method would pretty much crash with split or encoding errors if you pass invalid formats

Comment (Kami) [reply libcloud/common/gig_g8.py:60]: Thanks.

This should be fine for now.

With different forms of input strings, I meant thing such as JWT string payload missing some fields, being corrupted, etc. and making sure we handle that correctly.

**Comment by Kami** [comment]
Merged, thanks.

**Comment by Kami** [comment]
This required a bit more work to fix the tests failing in trunk - 4004375ccb9d5e899e70b85b33bb7a018c312eee

**Pull request merged**